### PR TITLE
Introduce slots and resolution of named variables

### PR DESCRIPTION
**Pull request by stefanoborini**
Introduces the concept of slots to data sources, and the application of named variables to each slot when running

### CODE REVIEW

File: force_bdss/core_evaluation_driver.py [hunk 1]
```diff
@@ -31,18 +31,96 @@ def application_started(self):
...
+
+    def _bind_data_values(self,
+                          available_data_values,
+                          model_slot_map,
+                          slots):
+
+        passed_data_values = []
+        lookup_map = {dv.name: dv for dv in available_data_values}
+
+        if len(slots) > len(model_slot_map):
```
Comment (martinRenou) [inline force_bdss/core_evaluation_driver.py]: Even if in the case of `len(slots) < len(model_slot_map)` it won't result in a failure, we should consider it a FileError, no ?

Comment (martinRenou) [inline force_bdss/core_evaluation_driver.py]: if `if len(slots) < len(model_slot_map)` it also means that there is an issue in the file, no ?

Comment (stefanoborini) [inline force_bdss/core_evaluation_driver.py]: I remember I had a valid reason for this, but I can't recall it now, so I guess you are right.

File: force_bdss/core_evaluation_driver.py [hunk 2]
```diff
@@ -31,18 +33,194 @@ def application_started(self):
...
+
+                logging.error(error_txt)
+                raise RuntimeError(error_txt)
+
+            if len(res) != len(ds_model.output_slot_names):
+                error_txt = (
+                    "The number of data values ({} values) returned"
+                    " by the DataSource '{}' does not match the number"
+                    " of user-defined names specified ({} values)."
+                    " This is likely a DataSource plugin error.").format(
```
Comment (martinRenou) [inline force_bdss/core_evaluation_driver.py]: `This is either an input file error or a plugin error` ?

File: force_bdss/core_plugins/dummy/kpi_adder/kpi_adder_calculator.py [hunk 3]
```diff
@@ -16,3 +17,15 @@ def run(self, model, data_source_results):
                 type=model.cuba_type_out,
                 value=sum
             )]
+
+    def slots(self, model):
+        return (
+            (
+                Slot(type=model.cuba_type_in),
```
Comment (martinRenou) [inline force_bdss/core_plugins/dummy/kpi_adder/kpi_adder_calculator.py]: We could already have a dynamic number of slots, depending on the model ? For the KPIAdder we could have a `model.number_inputs` integer

Comment (stefanoborini) [inline force_bdss/core_plugins/dummy/kpi_adder/kpi_adder_calculator.py]: I am planning to support that via variadic slots.

**Pull request merged**